httpx/_models.py

```diff
@@ -1007,6 +1023,8 @@ class Cookies(typing.MutableMapping[str, str]):
     HTTP Cookies, as a mutable mapping.
     """
 
+    jar: CookieJar
+
```

Comment:
Hrm. Restructuring `__init__` would probably stop this being ambiguous to `pyright`.

The `__init__` method should probably start with `self.jar = CookieJar()` before the branching cases. That'd seem easier to reason about from my perspective, as well as from `pyright`'s.